lib/fabrik/VirtualHostManager.js

```diff
@@ -0,0 +1,96 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const VirtualHostInstance = require('./VirtualHostInstance');
+const BaseManager = require('./BaseManager');
+const VirtualHostAgent = require('./VirtualHostAgent');
+const logger = require('../logger');
+const errors = require('../errors');
+const BadRequest = errors.BadRequest;
+const ServiceBindingAlreadyExists = errors.ServiceBindingAlreadyExists;
+const ServiceBindingNotFound = errors.ServiceBindingNotFound;
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+const NotFound = errors.NotFound;
+const mapper = require('./VirtualHostRelationMapper');
+
+class VirtualHostManager extends BaseManager {
+  constructor(plan) {
+    super(plan);
+    this.director = bosh.director;
+    this.agent = new VirtualHostAgent(this.settings.agent);
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  createVirtualHost(deploymentName, params, instanceId) {
+    logger.info(`Creating virtual host for deployment '${deploymentName}'...`);
+    return this.director.getDeploymentIps(deploymentName)
+      .then(ips => this.agent.createVirtualHost(ips, instanceId))
+      .then(() => this.mapper.createVirtualHostRelation(deploymentName, instanceId))
+      .tap(() => logger.info(`+-> Created virtual host for deployment '${deploymentName}'`));
+  }
+
+  deleteVirtualHost(deploymentName, instanceId) {
+    logger.info(`Deleting virtual host for deployment '${deploymentName}'...`);
+    let instanceDeleted = true;
+    return this.director.getDeploymentIps(deploymentName)
+      .tap(() => instanceDeleted = false)
+      .then(ips => Promise.all([this.agent.deleteVirtualHost(ips, instanceId),
+        this.mapper.deleteVirtualHostRelation(instanceId)
+      ]))
+      .tap(() => logger.info(`+-> Deleted virtual host for deployment '${deploymentName}'`))
+      .catch(NotFound, () => {
+        if (instanceDeleted) {
+          this.mapper.deleteVirtualHostRelation(instanceId);
```

Comment:
make this `return this.mapper.deleteVirtualHostRelation(instanceId);`

Reply:
If 'return' is added then it will not throw serviceInstanceNotFound and hence 410 gone, in case parent instance was already deleted.